paddle/framework/scope.cc

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/framework/scope.h"
+
+namespace paddle {
+namespace framework {
+
+Variable* Scope::CreateVariable(const std::string& name) {
+  if (!HasVariable(name)) {
```

Comment:
And local scope could create a variable that uses the same name in the parent.  It means the local variable can hide parent variable.

```
void CreateVariable(const std::string& name) {
  auto it = vars_.find(name);
  ASSERT_EQ(it, vars_.end());
  vars_[name] = std::unique_ptr<Variable>(new Variable());
}
```